### PR TITLE
Support new features for finding refereneces

### PR DESCRIPTION
**Pull request by norris-young**
1. Delete other windows when finding references, default t.
2. Open file in the original window if we do not delete other windows, default nil.

### CODE REVIEW

**Comment by norris-young** [comment]
I don't want to delete other windows when finding references and still need files being opened in the original buffer.
Add 2 customizable variable to make it available and keep the default behavior unchanged.

File: lsp-bridge-ref.el [hunk 1]
```diff
@@ -941,12 +967,24 @@ Function `move-to-column' can't handle mixed string of Chinese and English corre
...
   (when lsp-bridge-ref-window-configuration-before-search
-    (set-window-configuration lsp-bridge-ref-window-configuration-before-search)
+    ;; Restore window configuration before search if we delete-other-window before search.
+    ;; Or if we are inserting current line and the original window was deleted, we need to restore the window.
+    (if (or lsp-bridge-ref-delete-other-windows
+            (and insert
+                 (not (window-valid-p lsp-bridge-ref-requst-search-window))))
+        (set-window-configuration lsp-bridge-ref-window-configuration-before-search)
+      ;; Other we just delete the lsp-bridge-ref window, and select the original window.
+      (progn
```
Comment (manateelazycat) [inline lsp-bridge-ref.el:978]: Don't need progn in elisp else branch.

File: lsp-bridge-ref.el [hunk 2]
```diff
@@ -181,6 +195,9 @@ used to restore window configuration after finish search.")
   "Save buffer point before search,
 used to restore buffer point after finish search.")
 
+(defvar lsp-bridge-ref-requst-search-window nil
```
Comment (manateelazycat) [inline lsp-bridge-ref.el:198]: Should ```lsp-bridge-ref-request-search-window``` ? I found miss `e` in `request`.

File: lsp-bridge-ref.el [hunk 3]
```diff
@@ -941,12 +967,24 @@ Function `move-to-column' can't handle mixed string of Chinese and English corre
...
+        (delete-window)
+        (select-window lsp-bridge-ref-requst-search-window)))
     (goto-char lsp-bridge-ref-buffer-point-before-search)
     (setq lsp-bridge-ref-window-configuration-before-search nil)
-    (setq lsp-bridge-ref-buffer-point-before-search nil)))
+    (setq lsp-bridge-ref-buffer-point-before-search nil))
+  ;; clear saved window which requested search.
+  (if lsp-bridge-ref-open-file-in-requset-window
+      (setq lsp-bridge-ref-requst-search-window nil))
+  )
```
Comment (manateelazycat) [inline lsp-bridge-ref.el:987]: If i understand correctly, lsp-bridge-ref-request-search-window should reset to nil always, don't need ```if``` check here.

File: lsp-bridge-ref.el [hunk 4]
```diff
@@ -109,6 +109,20 @@ Default is enable, set this variable to nil if you don't like this feature."
...
   :group 'lsp-bridge-ref)
 
+(defcustom lsp-bridge-ref-delete-other-windows t
+  "Delete other windows while finding references.
+
+Default is enable, set this variable to nil if you don't like this feature."
+  :type 'boolean
+  :group 'lsp-bridge-ref)
+
+(defcustom lsp-bridge-ref-open-file-in-requset-window nil
```
Comment (manateelazycat) [inline lsp-bridge-ref.el:119]: Should ```lsp-bridge-ref-open-file-in-request-window``` ?

**Review by manateelazycat — CHANGES_REQUESTED** [review]

File: lsp-bridge-ref.el (hunk 1 above)
Comment (norris-young) [reply lsp-bridge-ref.el]: fixed

File: lsp-bridge-ref.el (hunk 2 above)
Comment (norris-young) [reply lsp-bridge-ref.el]: Fixed

File: lsp-bridge-ref.el (hunk 3 above)
Comment (norris-young) [reply lsp-bridge-ref.el]: Yeah, you are right.

File: lsp-bridge-ref.el (hunk 4 above)
Comment (norris-young) [reply lsp-bridge-ref.el]: Fixed

**Comment by norris-young** [comment]
Besides, I changed the argument name of `lsp-bridge-ref-quit` from `insert` to `restore`, which should be more suitable.

**Comment by manateelazycat** [comment]
Thanks for patch.

**Pull request merged**